vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml

```diff
@@ -234,32 +243,26 @@
       name: "{{ item.name }}"
       kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
     register: crd
-    vars:
-      query: "[?name == 'v1alpha3'].schema.openAPIV3Schema.description"
-    failed_when: crd.resources[0].spec.versions | json_query(query) is not search("{{ item.search }}")
+    failed_when: upgraded_capi_version not in (crd.resources[0].spec.versions | map(attribute='name'))
     loop:
       - name: machines.cluster.x-k8s.io
-        search: upgradedMachine
       - name: kubeadmcontrolplanes.controlplane.cluster.x-k8s.io
-        search: upgradedKubeadmControlPlane
       - name: kubeadmconfigs.bootstrap.cluster.x-k8s.io
-        search: upgradedKubeadmConfig
 
-  # Due to https://github.com/ansible-collections/kubernetes.core/issues/17
-  # we cannot use k8s_cluster_info. A fix is to be included in 2.0.0 of kubernetes.core
-  # - name: Verify upgraded API resource for Metal3Clusters
-  #   kubernetes.core.k8s_cluster_info:
-  #   register: api_status
-  #   # failed_when: api_status.apis not contains the upgraded m3c resource(s)?
+  - name: Verify upgraded API resources for CAPM3
+    kubernetes.core.k8s_cluster_info:
+      kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    register: api_status
+    failed_when: "'infrastructure.cluster.x-k8s.io/v1beta1' not in api_status.apis"
```

Comment:
is it possible to make `v1beta1` part as env var `UPGRADED_CAPM3_VERSION` or will it complicate things for future reader ?